community/index/src/main/java/org/neo4j/index/internal/gbptree/Header.java

```diff
@@ -48,7 +48,9 @@
 
     static final Writer CARRY_OVER_PREVIOUS_HEADER = (from,length,to) ->
     {
-        from.copyTo( from.getOffset(), to, to.getOffset(), length );
+        int toOffset = to.getOffset();
+        from.copyTo( from.getOffset(), to, toOffset, length );
+        to.setOffset( toOffset + length );
```

Comment:
Yes, `copyTo` will not actually move the `to` cursor, which means the header length is gonna be zero when we calculate it by calculating offset difference from before and after writing header.